### PR TITLE
Improves lesson 6-7 objectives

### PR DESCRIPTION
**Pull request by chrisbay**

### CODE REVIEW

**Comment by chrisbay** [comment]
[Java 7 objectives](https://trello.com/c/Kutijkh5/282-java-7-objectives)

File: unit2-java/lesson06.rst [hunk 1]
```diff
@@ -4,14 +4,18 @@ Unit 2 (Java) - Lesson 6 Learning Objectives
...
 
+- Understand the role of inheritance in enable effective code resuse.
+- Effectively use Java's inheritance tools to build clean, reusable code.
+
 Objectives
 ----------
 
 - Use the ``extends`` keyword to allow one class to inherit from another
-- Explain which fields and methods are inherited by a class extension, based on Java’s rules and taking into account access modifiers
+- Explain which fields and methods are inherited by a class extension, based on Java's rules and taking into account access modifiers
```
Comment (jimflores5) [inline unit2-java/lesson06.rst:14]: Just a general "why-am-I-finally-noticing-this" comment - to accommodate screen readers, we should convert the lists from ul to ol.

File: unit2-java/lesson06.rst [hunk 2]
```diff
@@ -4,14 +4,18 @@ Unit 2 (Java) - Lesson 6 Learning Objectives
...
 - Call a parent class constructor from a child class constructor
 - Override methods when extending a class
-- Describe the role of the ``Object`` class in each class’ hierarchy
+- Describe the role of the ``Object`` class in each class' hierarchy
 - Describe and use abstract classes and abstract methods
 - Explain how variable and parameter declaration relates to inheritance (i.e. how inheritance enables polymorphic behavior)
 - Use basic class diagrams to visualize and design class relationships
+- Write unit tests to test Java code that uses inheritance
```
Comment (jimflores5) [inline unit2-java/lesson06.rst]: "...tests to test..." kinda bothers me.  How about something like, "Write unit tests to verify Java code..." or, "Write tests for Java code..."

File: unit2-java/lesson06.rst [hunk 3]
```diff
@@ -4,14 +4,18 @@ Unit 2 (Java) - Lesson 6 Learning Objectives
 Goals
 -----
 
+- Understand the role of inheritance in enable effective code resuse.
```
Comment (jimflores5) [inline unit2-java/lesson06.rst]: Typo: "...in enabling effective..." or maybe, "...to enable effective..."

File: unit2-java/lesson06.rst [hunk 4]
```diff
@@ -4,14 +4,18 @@ Unit 2 (Java) - Lesson 6 Learning Objectives
...
+- Effectively use Java's inheritance tools to build clean, reusable code.
```
Comment (jimflores5) [inline unit2-java/lesson06.rst:8]: As phrased, this sounds like a repeat of the first objective.

File: unit2-java/lesson07.rst [hunk 5]
```diff
@@ -4,15 +4,19 @@ Unit 2 (Java) - Lesson 7 Learning Objectives
...
 -----
 
+- Understand how polymorphism is enabled by interfaces and inheritance.
+- Understand and use inheritance to create more flexible code. 
+
 Objectives
 ----------
 
-- Define polymorphism
+- Explain polymorphism and how it is useful
```
Comment (jimflores5) [inline unit2-java/lesson07.rst]: Maybe "how and why"?

File: unit2-java/lesson07.rst [hunk 6]
```diff
@@ -4,15 +4,19 @@ Unit 2 (Java) - Lesson 7 Learning Objectives
 Goals
 -----
 
+- Understand how polymorphism is enabled by interfaces and inheritance.
+- Understand and use inheritance to create more flexible code.
```
Comment (jimflores5) [inline unit2-java/lesson07.rst]: Did you mean "interfaces" instead of "inheritance"?

File: unit2-java/lesson07.rst [hunk 7]
```diff
@@ -4,15 +4,19 @@ Unit 2 (Java) - Lesson 7 Learning Objectives
...
+
 Objectives
 ----------
 
-- Define polymorphism
+- Explain polymorphism and how it is useful
 - Describe what an interface is and how it enables polymorphism
-- Explain whether an interface can be instantiated
+- Explain why it is preferable to use interface types in method declarations ("coding to an interface")
+- Explain why an interface can not be instantiated
```
Comment (jimflores5) [inline unit2-java/lesson07.rst]: Here, "cannot" instead of "can not".

File: unit2-java/lesson07.rst [hunk 8]
```diff
@@ -4,15 +4,19 @@ Unit 2 (Java) - Lesson 7 Learning Objectives
...
 - Implement common standard Java interfaces
-- Define what a default method is and whether you should write one when creating a new interface
+- Explain what a default method is and how they may be used with interfaces
 - Use the appropriate interface for collection types depending on the behavior they must exhibit
 - Describe the differences between abstract classes and interfaces
-- Explain what the benefits are to "coding to an interface"
 - Use the ``Comparator.compare()`` method
+- Write unit tests that test interface-based code
```
Comment (jimflores5) [inline unit2-java/lesson07.rst]: Similar to the unit 6 comment.

"...tests to test..." kinda bothers me. How about something like, "Write unit tests to verify..." or, "Write tests for interface-based code."

**Review by gildedgardenia — COMMENTED** [review]
Overall, I thought these looked great!

File: unit2-java/lesson06.rst (hunk 1 above)
Comment (chrisbaylewis) [reply unit2-java/lesson06.rst:14]: `ul` elements aren't incompatible with screen readers in and of themselves. The crucial point is to use `ol` and `ul` in the proper contexts. For example, for a table of contents with numbered chapters and sections `ol` should be used. The goals and objectives listed here aren't in any inherent or particular order, so I think `ul` is fine for this purpose.

File: unit2-java/lesson06.rst (hunk 4 above)
Comment (chrisbaylewis) [reply unit2-java/lesson06.rst:8]: I rephrased the first goal to be, "Understand the role of object-oriented inheritance to enable effective code resuse."

The distinction between this goal and the second is that the first points toward the more broad, language-agnostic concept of inheritance, while the second refers to the Java-specific syntax and tools used to implement inheritance.

Is it more clear now?

**Comment by chrisbay** [comment]
@jimflores5 I made changes to (hopefully) resolve each of your comments. Can you review the specific items left unresolved above and let me know what you think?

File: unit2-java/lesson06.rst (hunk 1 above)
Comment (jimflores5) [reply unit2-java/lesson06.rst:14]: OK.

File: unit2-java/lesson06.rst (hunk 4 above)
Comment (jimflores5) [reply unit2-java/lesson06.rst:8]: OK, the update makes the points more clear.

**Pull request merged**